operations/iuf/workflows/upgrade_all_products.md

```diff
@@ -1,120 +1,393 @@
+- Product content for all products found in `${MEDIA_DIR}` has been uploaded to the system
+- Product content uploaded to the system has been recorded in the product catalog
+- Per-stage product hooks have executed for the `process-media`, `pre-install-check`, and `deliver-product` stages
+
+### Perform manual product delivery operations
+
+**`NOTE`** This subsection is optional and can be skipped if third-party GPU and/or programming environment software is not needed.
 
-    (`ncn-m001#`) Prepare site variables file.
+Some products provide instructions for delivering third-party content to the system outside of IUF. If this content is desired, refer to the following documentation for instructions and execute the procedures before continuing
+with the workflow.
+
+- **Content:** Third-party GPU software
+  - **Description:** COS provides the `gpu-nexus-tool` script to upload third-party GPU software to Nexus. The GPU software is used later in the workflow when creating CFS configurations and building compute and application
+    node images.
+  - **Instructions:** See the "IUF Stage Details for COS" section of the [HPE Cray Operating System Installation Guide CSM on HPE Cray EX Systems (S-8025)](https://www.hpe.com/support/ex-S-8025) for references to the installation
+    procedures.
+- **Content:** Third-party programming environment software
+  - **Description:** CPE provides the `install-3p.sh` script to upload third-party programming environment software to Nexus. The programming environment software is used later in the workflow when creating CPE configurations
+    and building CPE images.
+  - **Instructions:** See the "CPE Install and Upgrade Framework usage" section of the [HPE CPE Installation Guide CSM on HPE Cray EX Systems (S-8003)](https://www.hpe.com/support/ex-S-8003) for references to the installation
+    procedures.
+
+1. Follow instructions to install third-party GPU software.
 
     ```bash
-    SITE_VARS_FILE=/etc/cray/upgrade/csm/iuf/site_vars.yaml
-    vi "${SITE_VARS_FILE}"
+    iuf -a "${ACTIVITY_NAME}" activity --create --comment "installing third-party GPU content" in_progress
+    < perform necessary operations per COS documentation >
+    iuf -a "${ACTIVITY_NAME}" activity --create waiting_admin
     ```
 
-1. Invoke `iuf run` to create an activity named `activity-all-products-upgrade` and use `-e` to execute the [`process-media`](../stages/process_media.md), [`pre-install-check`](../stages/pre_install_check.md),
-   [`deliver-product`](../stages/deliver_product.md), and [`update-vcs-config`](../stages/update_vcs_config.md) stages. Perform the upgrade using product content found in the `${ACTIVITY_DIR}` directory. Use site variables
-   from the `${SITE_VARS_FILE}` file and recipe variables from the `product_vars.yaml` file found in the `${ACTIVITY_DIR}/hpc-csm-software-recipe-23.05.0/vcs` `sat bootprep` configuration directory.
+1. Follow instructions to install third-party programming environment software.
```

Comment:
We should maybe mention the `iuf` command annotates the IUF time line with a note

Reply:
Per our conversation today, I'll revamp the use of `iuf activity` for timekeeping aspects and call this out in the new sections.

Reply:
Fixed in next update.